developers/node-tutorial.md

```diff
@@ -425,6 +425,157 @@ when you created the wallet.
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Blob was included at height %d\n", height)
+
+	// fetch the blob back from the network
+	retrievedBlobs, err := client.Blob.GetAll(ctx, height, []share.Namespace{namespace})
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Blobs are equal? %v\n", bytes.Equal(helloWorldBlob.Commitment, retrievedBlobs[0].Commitment))
+	return nil
+}
+```
+
+### Subscribing to new headers
+
+<!---
+Yet another thing: There is a argument rn that GetAll should return an error if no blobs are found. I do not agree with this argument, as it is not intuitive to the user, as seen in this example. I will try to resolve this before this PR is merged.
+--->
+
+You can subscribe to new headers using the [header.Subscribe](https://node-rpc-docs.celestia.org/?version=v0.11.0#header.Subscribe) method. This method returns a channel that will receive new headers as they are produced. In this example, we will fetch all blobs at the height of the new header in the `0xDEADBEEF` namespace.
+
+```go
+// SubscribeHeaders subscribes to new headers and fetches all blobs at the height of the new header in the 0xDEADBEEF namespace.
+func SubscribeHeaders(ctx context.Context, url string, token string) error {
+	client, err := client.NewClient(ctx, url, token)
+	if err != nil {
+		return err
+	}
+
+	// create a namespace to filter blobs with
+	namespace, err := share.NewBlobNamespaceV0([]byte{0xDE, 0xAD, 0xBE, 0xEF})
+	if err != nil {
+		return err
+	}
+
+	// subscribe to new headers using a <-chan *header.ExtendedHeader channel
+	headerChan, err := client.Header.Subscribe(ctx)
+	if err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case header := <-headerChan:
+			// fetch all blobs at the height of the new header
+			blobs, err := client.Blob.GetAll(context.TODO(), header.Height(), []share.Namespace{namespace})
+			if err != nil {
+				fmt.Printf("Error fetching blobs: %v\n", err)
+			}
+
+			fmt.Printf("Found %d blobs at height %d in 0xDEADBEEF namespace\n", len(blobs), header.Height())
+		case <-ctx.Done():
+			return nil
+		}
+	}
+}
+```
+
+### Fetching an Extended Data Square (EDS)
+
+You can fetch an [Extended Data Square (EDS)](https://celestiaorg.github.io/celestia-app/specs/data_structures.html#erasure-coding) using the [share.GetEDS](https://node-rpc-docs.celestia.org/?version=v0.11.0#share.GetEDS) method. This method takes a header and returns the EDS at the given height.
+
+```go
+// GetEDS fetches the EDS at the given height.
+func GetEDS(ctx context.Context, url string, token string, height uint64) (*rsmt2d.ExtendedDataSquare, error) {
+	client, err := client.NewClient(ctx, url, token)
+	if err != nil {
+		return nil, err
+	}
+
+	// First get the header of the block you want to fetch the EDS from
+	header, err := client.Header.GetByHeight(ctx, height)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fetch the EDS
+	return client.Share.GetEDS(ctx, header)
+}
+```
+
+### Balances and Transfers
+
+TODO
+
```

Comment:
until here, could be moved into new doc. what do you think?

then I will tidy up / cut down the existing CLI tutorial